src/hw1/bash_imitating_functions.py

```diff
@@ -0,0 +1,58 @@
+def _read_file(filename):
+    try:
+        file = open(filename)
+    except IOError:
+        print("No such file or directory")
+        return None
+    except:
+        print("Unexpected error")
+        return None
+
+    data = file.read()
+    file.close()
+    return data
```

Comment:
This is actually a rather poor way to do it. The problem here is that for 3 out of 4 functions (head, wc, and nl) you can through files iteratively. If you try doing the work on a 10GB file, your way would load the entire file. Even if you need to read just first 10 lines and that's it. 

So I would advise doing the task without loading everything (except for tail)